src/models/Location.js

```diff
@@ -15,6 +15,7 @@ import { isPlainObject, getScript, makeAsyncRequest } from "../utils"
 export default class Location extends Base {
   constructor( options ) {
     super( defaultOptions, options )
+    this.options = { ...defaultOptions, ...options }; // combine default options and user options
```

Comment:
```suggestion
    this.options = { ...defaultOptions, ...(options || {}) }; // combine default options and user options
```

I'm good with that as the `super`s merging behavior (https://github.com/sandstreamdev/cookieconsent/blob/dev/src/models/Base.js#L10) is inappropriate for this case. I just wonder why this [particular test case passes](https://github.com/sandstreamdev/cookieconsent/blob/dev/src/models/Location.spec.js#L33) when it should fail when options are missing (object spread of `undefined`)? 🤔

Reply:
Thanks for this!

I've gone and updated my PR with your suggestion, as well as added another test case to cover passing custom `serviceDefinitions` options. This should help catch this sort of error in the future 😄 

From what I can tell, the existing test case was only validating the structure of the object (`isLocation`) and the merged `options` object keys (`hasCorrectOptions`).

With the way `Locations.js` is written, the `getServiceByIdx()` function needs to be called to actually cause an error. The error is caused when it goes to access `this.options.serviceDefinitions[...]`, and the custom object key does not exist.